perf/smallbank_workload/src/main.rs

```diff
@@ -344,6 +479,58 @@ fn run_playlist_process_command(args: &ArgMatches) -> Result<(), Box<Error>> {
     Ok(())
 }
 
+/// Transforms SmallbankTransactionPayloads into Sawtooth Transactions.
+pub struct SBPayloadTransformer<'a> {
+    signer: &'a signing::Signer<'a>
+}
+
+impl<'a> SBPayloadTransformer<'a> {
+    pub fn new(signer: &'a signing::Signer) -> Self {
+        SBPayloadTransformer {
+            signer: signer,
+        }
+    }
+
+    pub fn payload_to_transaction(&self, payload: SmallbankTransactionPayload)
+        -> Result<Transaction, Box<Error>>
+    {
+        let mut txn = Transaction::new();
+        let mut txn_header = TransactionHeader::new();
+
+        txn_header.set_family_name(String::from("smallbank"));
+        txn_header.set_family_version(String::from("1.0"));
+
+        let elapsed = Instant::now().elapsed();
+        txn_header.set_nonce(format!("{}{}", elapsed.as_secs(), elapsed.subsec_nanos()));
+
+        let addresses = protobuf::RepeatedField::from_vec(make_addresses(&payload));
+
+        txn_header.set_inputs(addresses.clone());
+        txn_header.set_outputs(addresses.clone());
+
+        let payload_bytes = try!(payload.write_to_bytes());
+
+        let mut sha = Sha512::new();
+        sha.input(&payload_bytes);
+        let hash: &mut [u8] = & mut [0; 64];
+        sha.result(hash);
+
+        txn_header.set_payload_sha512(bytes_to_hex_str(hash));
+        txn_header.set_signer_public_key(self.signer.get_public_key()?.as_hex());
+        txn_header.set_batcher_public_key(self.signer.get_public_key()?.as_hex());
+
+        let header_bytes = try!(txn_header.write_to_bytes());
+
+        let signature = try!(self.signer.sign(&header_bytes.to_vec()));
```

Comment:
Use `?;` instead of `try!()`: https://doc.rust-lang.org/std/macro.try.html

Reply:
Thanks, @aludvik . I will fix those try!s and https://jira.hyperledger.org/browse/STL-1014 here is the Jira story.